jupyterhub_moss/form/option_form.js

```diff
@@ -165,6 +257,18 @@ function restoreConfigFromLocalStorage() {
     jupyterlabSimpleElem.checked = config['fields']['jupyterlab'];
     jupyterlabSimpleElem.dispatchEvent(new Event('change'));
   }
+
+  // Restore Jupyter environment
+  const environment = config['environment'];
+  if (environment === undefined || (environment['isCustom'] && !config['isAdvanced'])) {
+    // Do not restore environment if it was not saved or if custom was saved for simple tab.
+    updateEnvironmentSelect();
+  } else {
+    if (environment['isCustom']) {
+      environmentCustomOptionElem.value = environment['value'];
```

Comment:
Current value of custom path is stored as custom option value.
I'd take any better way.